### ISSUE TITLE
TF Slim - allow soft placement for devices with train_image_classifier

### ISSUE THREAD

**Issue by vade** [tensorflow/tensorflow#11187]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: No
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: macOS 10.12.5
- **TensorFlow installed from (source or binary)**:source
- **TensorFlow version (use command below)**: git tag 1.2
- **Python version**: 2.7 system install
- **Bazel version (if compiling from source)**: home-brew 0.4.5
- **CUDA/cuDNN version**: NA
- **GPU model and memory**:  NA
- **Exact command to reproduce**: 

train Mobilenet_v1 on CPU like so:

 python train_image_classifier.py     --train_dir=${TRAIN_DIR}     --dataset_dir=${DATASET_DIR}     --dataset_name=Framing     --dataset_split_name=train     --model_name=mobilenet_v1     --checkpoint_path=${CHECKPOINT_PATH}     --checkpoint_exclude_scopes=MobilenetV1/Logits/Conv2d_1c_1x1/biases,MobilenetV1/Logits/Conv2d_1c_1x1/weights 

### Describe the problem
Describe the problem clearly here. Be sure to convey here why it's a bug in TensorFlow or a feature request.

Tensorflow 1.2 has no GPU support for macOS. Thus training/retraining can only happen on CPU.
TF Slim doesnt appear to have an out of the box way to specify soft placement of nodes - therefore I can't appear to train a mobile net checkpoint?

### Source code / logs
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached. Try to provide a reproducible test case that is the bare minimum necessary to generate the problem.

`INFO:tensorflow:Summary name /clone_loss is illegal; using clone_loss instead.
INFO:tensorflow:Fine-tuning from /Volumes/MediaArchive/datasets/SynopsisCinemaNet/model/mobilenet_v1_1.0_224_2017_06_14/mobilenet_v1_1.0_224.ckpt.index
INFO:tensorflow:Error reported to Coordinator: <class 'tensorflow.python.framework.errors_impl.InvalidArgumentError'>, Cannot assign a device for operation 'MobilenetV1/Logits/Conv2d_1c_1x1/biases/RMSProp_1': Operation was explicitly assigned to /device:GPU:0 but available devices are [ /job:localhost/replica:0/task:0/cpu:0 ]. Make sure the device specification refers to a valid device.
	 [[Node: MobilenetV1/Logits/Conv2d_1c_1x1/biases/RMSProp_1 = VariableV2[_class=["loc:@MobilenetV1/Logits/Conv2d_1c_1x1/biases"], container="", dtype=DT_FLOAT, shape=[5], shared_name="", _device="/device:GPU:0"]()]]
`

**Comment by vade** [comment]
So, spending some time getting more familiar with Slim, it appears that commending out some specific code in model_deploy where clone_on_cpu is checked calls into setting the GPU device has made it work. 

It would be nice for this script to auto-detect available devices and not assume GPU support since a shipping, supported platform does not provide CPU support.

Thank you in advance for consideration.

**Comment by andydavis1** [comment]
@sguada can you comment on this one?

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting tensorflower` label was assigned. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting tensorflower` label was assigned. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Awaiting TensorFlower: It has been 14 days with no activity and the `awaiting tensorflower` label was assigned. Please update the label and/or status accordingly.

**Comment by xxllp** [comment]
???

**Comment by sguada** [comment]
You can configure using FLAGS.clone_on_cpu=True
This issue should be under https://github.com/tensorflow/models/issues

**Comment by kbhardwa** [comment]
Any solution to this? Just naively doing FLAGS.clone_on_cpu=True makes it use only CPU as workers.
